frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationContext.tsx

```diff
@@ -41,19 +50,31 @@ const defaultValue: CalibrationContextType = {
 	isCalibrating: false,
 	origin: '',
 	setOrigin: () => {},
-	arrow2D: { start: null, end: null },
+	isCalibrating3D: false,
+	setIsCalibrating3D: () => {},
+	vector3D: EMPTY_VECTOR,
+	setVector3D: () => {},
+	resetVector3D: () => {},
+	arrow2D: EMPTY_VECTOR,
 	setArrow2D: () => {},
+	drawingId: '',
 };
 export const CalibrationContext = createContext(defaultValue);
 CalibrationContext.displayName = 'CalibrationContext';
 
 export const CalibrationContextComponent = ({ children }) => {
 	const { teamspace, project, revision, containerOrFederation } = useParams();
-	const [step, setStep] = useState(2);
+	const [step, setStep] = useState(0);
 	const [isStepValid, setIsStepValid] = useState(false);
 	const [origin, setOrigin] = useState(generatePath(DRAWINGS_ROUTE, { teamspace, project }));
 	const [isCalibrating] = useSearchParam('isCalibrating', Transformers.BOOLEAN);
-	const [arrow2D, setArrow2D] = useState({ start: null, end: null });
+	const [isCalibrating3D, setIsCalibrating3D] = useState(false);
+	const [vector3D, setVector3D] = useState<Vector3D>(EMPTY_VECTOR);
+	const [arrow2D, setArrow2D] = useState(EMPTY_VECTOR);
+	const [drawingId] = useSearchParam('drawingId');
+	const drawing = DrawingsHooksSelectors.selectDrawingById(drawingId);
+
+	const resetVector3D = () => setVector3D(drawing?.vector3D || EMPTY_VECTOR);
```

Comment:
I dont think is necessary to have resetVector3d here. I would move that to the calibration3dHandler